project_task_code/models/project_task.py

```diff
@@ -42,3 +43,14 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def name_search(self, name="", args=None, operator="ilike", limit=100):
+        """Allow searching by code by default."""
+        if name:
+            args = (
+                expression.OR([[("code", operator, name)], args])
+                if args
+                else ["|", ("code", operator, name)]
```

Comment:
This is an incomplete term because it's a binary operator followed by only one operand.

Reply:
Probably happening to you because you're doing a search without a term (a.k.a. "give me all results"). Then you hit https://github.com/odoo/odoo/blob/8f115ccc20f93f52a93dd6836b021463e7227c91/odoo/models.py#L1927 which probably triggers that bug.

Reply:
Yeah, using the unittesting is working well and I have tested from odoo-bin shell

thank you!

So, approve?